Controller/TemplateController.php

```diff
@@ -15,39 +15,28 @@
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\Serializer\Serializer;
-use Symfony\Component\Serializer\Encoder\JsonEncoder;
-use Symfony\Component\Serializer\Normalizer\GetSetMethodNormalizer;
-use CampaignChain\CoreBundle\Entity\Module;
 use CampaignChain\CoreBundle\Entity\Action;
 
+/**
+ * Class TemplateController
+ * @package CampaignChain\Campaign\TemplateBundle\Controller
+ */
 class TemplateController extends Controller
 {
     const CAMPAIGN_DISPLAY_NAME = "Campaign Template";
     const BUNDLE_NAME = 'campaignchain/campaign-template';
     const MODULE_IDENTIFIER = 'campaignchain-template';
     const TRIGGER_HOOK = 'campaignchain-timespan';
 
-    public function indexAction(){
+    public function indexAction()
+    {
         // Get the campaign templates
-        $qb = $this->getDoctrine()->getEntityManager()->createQueryBuilder();
-        $qb->select('c')
-            ->from('CampaignChain\CoreBundle\Entity\Campaign', 'c')
-            ->from('CampaignChain\CoreBundle\Entity\Module', 'm')
-            ->from('CampaignChain\CoreBundle\Entity\Bundle', 'b')
-            ->where('b.name = :bundleName')
-            ->andWhere('m.identifier = :moduleIdentifier')
-            ->andWhere('m.id = c.campaignModule')
-            ->setParameter('bundleName', static::BUNDLE_NAME)
-            ->setParameter('moduleIdentifier', static::MODULE_IDENTIFIER)
-            ->orderBy('c.name', 'ASC');
-        $query = $qb->getQuery();
-        $repository_campaigns = $query->getResult();
+        $repository_campaigns = $this->getDoctrine()->getRepository('CampaignChainCoreBundle:Campaign')->getCampaignsByModule(static::MODULE_IDENTIFIER);
```

Comment:
I analyzed the SQL queries and the bundle table is never used. If you check the old query above - "bundle - b" is imported and filtered, but there is no connection between the bundle and the campaign. Only the module is relevant. 
It's like adding a random table to the query - it has no effect on the result.
If it should have a effect, then the old query was doing something wrong. Please correct me, if I'm missing something :)